src/overlays/actors/ovl_Bg_Ikana_Dharma/z_bg_ikana_dharma.c

```diff
@@ -28,47 +34,229 @@ const ActorInit Bg_Ikana_Dharma_InitVars = {
+        s32 temp_v0_2;
+        g_firstHitBgIkanaDharma = this;
+        Flags_SetSwitch(globalCtx, BGIKANADHARMA_GET_SWITCHFLAG(&this->dyna.actor));
+        temp_v0_3 = (s16)(this->dyna.actor.yawTowardsPlayer + 0x8000);
+        temp_v0_2 = ((s16)(temp_a3->shape.rot.y - temp_v0_3) >> 1);
+        this->dyna.actor.world.rot.y = temp_v0_3 + temp_v0_2 + 0xF000;
+        this->dyna.actor.speedXZ = 20.0f;
+        Actor_PlaySfxAtPos(&this->dyna.actor, NA_SE_EV_DARUMA_VANISH);
+        BgIkanaDharma_BeginWaitToStartCutscene(this);
+    } else if ((this->dyna.actor.flags & 0x40) == 0x40 && g_firstHitBgIkanaDharma == NULL &&
+               this->dyna.actor.xzDistToPlayer < 420.0f) {
+        temp_v0_3 = (s16)(this->dyna.actor.yawTowardsPlayer - temp_a3->shape.rot.y);
+        if (temp_v0_3 < 0) {
+            temp_v0_3 = -temp_v0_3; // this line is the reason temp_v0_3 isn't s16
+        }
+        if (temp_v0_3 >= 0x4001) {
+            Collider_UpdateCylinder(&this->dyna.actor, &this->collider);
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &thisx->collider.base);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC7EC.s")
+void BgIkanaDharma_BeginWaitToStartCutscene(BgIkanaDharma* this) {
+    ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+    this->actionFunc = BgIkanaDharma_WaitToStartCutscene;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC808.s")
+void BgIkanaDharma_WaitToStartCutscene(BgIkanaDharma* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCanPlayNext(this->dyna.actor.cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->dyna.actor.cutscene, &this->dyna.actor);
+        BgIkanaDharma_BeginCutsceneState(this);
+    } else {
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/BgIkanaDharma_Update.s")
+void BgIkanaDharma_BeginCutsceneState(BgIkanaDharma* this) {
+    this->cutsceneFramesRemaining = 0xA;
+    this->actionFunc = &BgIkanaDharma_UpdateCutscene;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/BgIkanaDharma_Draw.s")
+void BgIkanaDharma_UpdateCutscene(BgIkanaDharma* this, GlobalContext* globalCtx) {
+    if (this->cutsceneFramesRemaining > 0) {
+        this->cutsceneFramesRemaining--;
+        if (this->cutsceneFramesRemaining == 0) {
+            if (g_firstHitBgIkanaDharma == this) {
+                g_firstHitBgIkanaDharma = NULL;
+            }
+            ActorCutscene_Stop(this->dyna.actor.cutscene);
+        }
+    }
+    if (Math_StepToF(&this->dyna.actor.scale.y, 0.0f, 1.0f / 300.0f) != 0) {
+        Actor_MarkForDeath(&this->dyna.actor);
+        return;
+    }
+    this->dyna.actor.scale.x = this->dyna.actor.scale.y * 3.0f;
+    this->dyna.actor.scale.z = this->dyna.actor.scale.y * 3.0f;
+    BgIkanaDharma_CreateParticles(this, globalCtx);
```

Comment:
```suggestion
    if (Math_StepToF(&this->dyna.actor.scale.y, 0.0f, 1.0f / 300.0f) != 0) {
        Actor_MarkForDeath(&this->dyna.actor);
    } else {
        this->dyna.actor.scale.x = this->dyna.actor.scale.y * 3.0f;
        this->dyna.actor.scale.z = this->dyna.actor.scale.y * 3.0f;
        BgIkanaDharma_CreateParticles(this, globalCtx);
    }
```
This is optional, just wanted to point out when you see `if (return) other` it can be turned into `if else`. This is mostly style and you can go either way.

Reply:
Most of the time yes, but `MarkForDeath` is one of the exceptions.